### PR TITLE
Merge release7x into release

### PR DESCRIPTION
**Pull request by big-guy**

### CODE REVIEW

File: build-logic-commons/commons/src/main/kotlin/common.kt [hunk 1]
```diff
@@ -20,6 +20,5 @@ import org.gradle.jvm.toolchain.JvmVendorSpec
 fun JavaPluginExtension.configureJavaToolChain() {
     toolchain {
         languageVersion.set(JavaLanguageVersion.of(11))
-        vendor.set(JvmVendorSpec.ADOPTIUM)
```
Comment (wolfs) [inline build-logic-commons/commons/src/main/kotlin/common.kt:23]: I don't think this should be removed.

File: subprojects/soak/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadComplexProjectSoakTest.groovy [hunk 2]
```diff
@@ -17,9 +17,10 @@
 package org.gradle.jvm.toolchain
 
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
 import spock.lang.Ignore
 
-class JavaToolchainDownloadComplexProjectSoakTest extends AbstractIntegrationSpec {
+class JavaToolchainDownloadComplexProjectSoakTest  extends AbstractIntegrationSpec {
```
Comment (wolfs) [inline subprojects/soak/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadComplexProjectSoakTest.groovy:23]: It should be better without the ` `. Why do you re-add it here?

File: build-logic-commons/commons/src/main/kotlin/common.kt (hunk 1 above)
Comment (big-guy) [reply build-logic-commons/commons/src/main/kotlin/common.kt:23]: @hegyibalint just removed this on release7x: https://github.com/gradle/gradle/commit/1d3e8ed9d6d4bd855962bf60e2aaf82b4798a91b

Comment (ljacomet) [reply build-logic-commons/commons/src/main/kotlin/common.kt:23]: Well, that removal is a mistake. It looks like the intent was to remove the conditional, not the vendor. The vendor should stay.

File: subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadIntegrationTest.groovy [hunk 3]
```diff
@@ -137,32 +137,68 @@ class JavaToolchainDownloadIntegrationTest extends AbstractIntegrationSpec {
...
-
-            public abstract class CustomToolchainResolver implements JavaToolchainResolver {
-                @Override
-                public Optional<JavaToolchainDownload> resolve(JavaToolchainRequest request) {
-                    URI uri = URI.create("http://exoticJavaToolchain.com/java-" + request.getJavaToolchainSpec().getLanguageVersion().get());
-                    return Optional.of(JavaToolchainDownload.fromUri(uri));
-                }
-            }
-            
+        """
```
Comment (ljacomet) [inline subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadIntegrationTest.groovy:140]: There should be no changes in this file.

File: subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadSpiIntegrationTest.groovy [hunk 4]
```diff
@@ -369,6 +369,85 @@ class JavaToolchainDownloadSpiIntegrationTest extends AbstractJavaToolchainDownl
         failure.getOutput().contains("Explicitly requested toolchains: [useless1, useless3].")
     }
 
+    def "cannot mutate repository rules after settings have been evaluated"() {
```
Comment (ljacomet) [inline subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadSpiIntegrationTest.groovy:372]: This seems to introduce duplicate tests, there should be no changes to this file

**Review by ljacomet — CHANGES_REQUESTED** [review]

File: build-logic-commons/commons/src/main/kotlin/common.kt (hunk 1 above)
Comment (big-guy) [reply build-logic-commons/commons/src/main/kotlin/common.kt:23]: OK, I put that back in here https://github.com/gradle/gradle/pull/23243

**Review by ljacomet — APPROVED** [review]

**Comment by big-guy** [comment]
@bot-gradle test and merge

File: build-logic-commons/commons/src/main/kotlin/common.kt (hunk 1 above)
Comment (wolfs) [reply build-logic-commons/commons/src/main/kotlin/common.kt]: Yep, should be like now on `master`: https://github.com/gradle/gradle/blob/9842b0fa16ad451a449b88bde9e8647d3da84756/build-logic-commons/commons/src/main/kotlin/common.kt#L23

Do we need to change this on `release7x` as well then?

**Review by wolfs — APPROVED** [review]

**Pull request merged**